syntax = "proto3";

service StockMarket {
    rpc GetStock (GetStockRequest) returns (GetStockResponse);
}

message GetStockRequest {
    string symbol = 1;
}

message GetStockResponse {
    Stock stock = 1;
}

message Stock {
    string symbol = 1;
    string full_name = 2;
    float current_price = 3;
    float current_pe_ratio = 4;
}